arbor/partition_load_balance.cpp

```diff
@@ -53,11 +65,81 @@ domain_decomposition partition_load_balance(
 
     // Local load balance
 
-    std::unordered_map<cell_kind, std::vector<cell_gid_type>> kind_lists;
+    std::vector<std::vector<cell_gid_type>> super_cells; //cells connected by gj
+    std::vector<cell_gid_type> reg_cells; //independent cells
+
+    // Map to track visited cells (cells that already belong to a group)
+    std::unordered_map<cell_gid_type, bool> visited;
```

Comment:
Should be able to use `std::unordered_set<cell_gid_type>` instead.

Reply:
Yes, that does look much nicer. I think this section went too many iterations and I should have just started thinking about it from scratch. Thank you.